src/hooks/useRepositoriesList.ts

```diff
@@ -1,12 +1,36 @@
 import useSWR from "swr";
+import { supabase } from "../lib/supabase";
 
 interface PaginatedRepoResponse {
   readonly data: DbRepo[];
   readonly meta: PageMetaDto;
 }
 
+const apiFetcherWithToken = async (apiUrl: string, token: string) => {
+  const res = token
+    ? await fetch(`${import.meta.env.VITE_API_URL}/${apiUrl}`, { headers: { accept: "application/json", Authorization: `Bearer ${token}` } })
+    : await fetch(`${import.meta.env.VITE_API_URL}/${apiUrl}`, { headers: { accept: "application/json" } });
+
+  if (!res.ok) {
+    const error = new Error("HttpError");
+
+    error.message = `${res.status} ${res.statusText}`;
+    error.stack = JSON.stringify(await res.json());
+
+    throw error;
+  }
+
+  return res.json();
+};
+
 const useRepositoriesList = (orderBy = "stars", limit = 10) => {
```

Comment:
this is not correct - we will have 4 types of voting mechanisms, we cannot `switch` between them in a repos list; this requires a dedicated hook like `useVotedRepositories.ts`

Reply:
committed